docs/source/production.rst

```diff
@@ -85,3 +85,52 @@ such as:
   lines). The default is `origin/master`.
 * Configuring `diff-quality` to return an error code if the quality is too low
 * Troubleshooting
+
+Using `pre-commit`_
+^^^^^^^^^^^^^^^^^^^
+
+`pre-commit`_ is a framework to manage git "hooks"
+triggered right before a commit is made.
+
+A `git hook`_ is a git feature to "fire off custom scripts"
+when specific actions occur.
+
+Using `pre-commit`_ with SQLFluff is a good way
+to provide automated linting to SQL developers.
+
+With `pre-commit`_, you also get the benefit of
+only linting/fixing the files that changed.
+
+SQLFluff comes with two `pre-commit`_ hooks:
+
+* sqlfluff-lint: returns linting errors.
+* sqlfluff-fix: attempts to fix rule violations.
+
+You should create a file named `.pre-commit-config.yaml`
+at the root of your git project, which should look
+like this:
+
+.. code-block:: yaml
+
+  repos:
+  - repo: https://github.com/sqlfluff/sqlfluff
+    rev: v|release|
```

Comment:
IIUC, the `v` should be removed. It looks like `pre-commit` attempts to check out a tag or branch with this name, but SQLFluff branches thus far do not use a `v` prefix, e.g.:

```
git tag --list
0.0.1
0.0.2
0.0.3
0.0.6
0.0.7
0.1.0
0.1.4
0.1.5
0.2.0
0.2.1
0.2.2
0.2.3
0.2.4
0.3.0
0.3.1
0.3.2
0.3.3
0.3.4
0.3.5
0.3.6
```

Reply:
This worked for me:
```
-   repo: https://github.com/sqlfluff/sqlfluff
    rev: master
    hooks:
    -   id: sqlfluff-lint
    -   id: sqlfluff-fix
```

Of course, once there's a release with the pre-commit hooks, `rev:` should point to that, not `master`. I was just trying to get this working for one of our projects to try it out. This is great stuff! I've been wanting pre-commit support for a long time.

Reply:
@barrywhart thanks for pointing out the problem with `v` in the docs, the pre-commit I had set up with the "mirror" added `v` as a prefix to releases so I made that mistake in the docs.
https://github.com/sqlfluff/sqlfluff/pull/611